alica_engine/src/engine/collections/Variant.cpp

```diff
@@ -3,7 +3,41 @@
 namespace alica
 {
 
-constexpr size_t Variant::kUnionSize;
-constexpr size_t Variant::kVariantSize;
+void variant::serializeTo(std::string& arr, const Variant& var)
+{
+    arr.clear();
+    arr = arr + std::to_string(var.index()); // type
+
+    std::visit( [&](auto&& arg) {
+        using T = std::decay_t<decltype(arg)>;
+        if constexpr (!(std::is_same_v<std::monostate, T> || std::is_same_v<void*, T>)) {
```

Comment:
is void* used anywhere?

Reply:
I couldn't find anywhere that uses it currently.
I kinda want to remove it because atm we're not properly handling it anyway

Reply:
As far as I can tell we are not using anything but double.  Definitely we should unsupport void* imo

Reply:
ok i'm removing it